packages/server/src/app/db/content/blogs/blogs.service.ts

```diff
@@ -30,7 +33,12 @@ export class BlogsService {
             'stats.words': await countPlaintextWords(await stripAllHtml(blogInfo.body))
         });
 
-        return await newBlog.save();
+        const savedBlog = await newBlog.save();
+
+        // Subscribe the author to comments on their new blog
+        await this.notificationsService.subscribe(user.sub, savedBlog._id, NotificationKind.CommentNotification);
```

Comment:
...and this is why `sourceId` has to be the ID of the blog. Authors will get notifications for any `CommentKind` notifications with from a source of <their new blog's ID>.